### PR TITLE
Refactor license constants and utils

### PR DESCRIPTION
**Pull request by dhruvkb**
## Description

This PR refactors and cleans up in the implementation of license constants and utils and their usage in the code.

This PR increases the code's parity with the frontend. A similar effort on the frontend is happening in https://github.com/WordPress/openverse-api/pull/685.

## Testing Instructions

Since this is a refactor, there are no changes to test but rather the test is that there should be no changes.

0. Check out the PR and run the dev server.
1. Visit the single result page and see that all 3 of the attribution tabs show the same content as before.
2. Try to copy the attribution using the copy button, it should be copied correctly to the clipboard.
3. Verify by pasting in a rich text editor like WordPad on Windows or TextEdit in RTF mode on macOS.
4. Try pasting HTML in [an HTML viewer](onlinehtmleditor.net/). You should see the hyperlinks and icons.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by sarayourfriend** [comment]
@dhruvkb can you add testing instructions (and fill out the checklist)? Thanks!

**Comment by dhruvkb** [comment]
@sarayourfriend done!

**Review by sarayourfriend — APPROVED** [review]
LGTM! The visual regression tests under the "media-reuse" tests also confirm there are no changes to the output :tada:

File: src/models/media.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export interface Media {
 
   license: License
   license_version: LicenseVersion
-  license_url: string
+  license_url?: string
```
Comment (krysal) [inline src/models/media.ts:24]: Why does the `licensr_url` change to an optional field?

Comment (dhruvkb) [reply src/models/media.ts:24]: Practically it's never undefined, I added that just to ensure that things work even if it was undefined.

File: test/unit/specs/utils/attribution-html.spec.ts [hunk 2]
```diff
@@ -78,6 +78,15 @@ describe('getAttribution', () => {
     expect(attrText).toEqual(expectation)
   })
 
+  it('skips the link if URL is missing', () => {
+    const mediaItemNoLicenseUrl = { ...mediaItem, license_url: undefined }
+    const attrText = getAttribution(mediaItemNoLicenseUrl, i18n, {
+      isPlaintext: true,
+    })
+    const antiExpectation = 'To view'
```
Comment (obulat) [inline test/unit/specs/utils/attribution-html.spec.ts:86]: Love the name :)

**Review by obulat — COMMENTED** [review]
Would it be possible to add a different language to the tests?

**Comment by dhruvkb** [comment]
@obulat I didn't get what you mean by different language. Are you referring to i18n functionality or just having the values be in a different language?

**Comment by dhruvkb** [comment]
@obulat if you mean like for testing i18n, that'll be difficult because we only have the `en.json` file tracked in the code. All the other languages are retrieved dynamically.

**Comment by obulat** [comment]
Oh, that's true!
It might be nice to mock it.
But it's not a blocker.

**Review by obulat — APPROVED** [review]

**Pull request merged**